opencensus/common/resource.py

```diff
@@ -124,3 +147,57 @@ def merge(self, other):
         :return: The new combined resource.
         """
         return merge_resources(self, other)
+
+
+def unquote(string):
+    """Strip quotes surrounding `string` if they exist.
+
+    >>> unquote('abc')
+    'abc'
+    >>> unquote('"abc"')
+    'abc'
+    >>> unquote("'abc'")
+    'abc'
+    >>> unquote('"a\\'b\\'c"')
+    "a'b'c"
+    """
+    return UNQUOTE_RE.sub(r'\2', string)
+
+
+def parse_labels(labels_str):
+    """Parse label keys and values following the Resource spec.
+
+    >>> parse_labels("k=v")
+    {'k': 'v'}
+    >>> parse_labels("k1=v1, k2=v2")
+    {'k1': 'v1', 'k2': 'v2'}
+    >>> parse_labels("k1='v1,=z1'")
+    {'k1': 'v1,=z1'}
+    """
+    if not LABELS_RE.match(labels_str):
+        return None
+    labels = {}
+    for kv in KV_RE.finditer(labels_str):
+        gd = kv.groupdict()
+        labels[unquote(gd['key'])] = unquote(gd['val'])
```

Comment:
Do we want to check for duplicated keys and throw error here?

Reply:
Sure, added in 694ed5a.

Reply:
In other languages if there're duplicated key-value pairs, we simply overwrite with the last value. May be better to be consistent here.

Reply:
Changed to log a warning instead in a265e0e, which I think is a good compromise.